File: src/main/java/io/stouder/adonis/edge/EdgeFileElementType.java
package io.stouder.adonis.edge;

import com.intellij.psi.impl.source.html.HtmlFileImpl;
import com.intellij.psi.stubs.PsiFileStub;
import com.intellij.psi.tree.IStubFileElementType;

public class EdgeFileElementType extends IStubFileElementType<PsiFileStub<HtmlFileImpl>> {
    public static final IStubFileElementType<PsiFileStub<HtmlFileImpl>> INSTANCE  = new EdgeFileElementType();

    private EdgeFileElementType() {
        super("edge", EdgeLanguage.INSTANCE);
    }
}

File: src/main/java/io/stouder/adonis/edge/EdgeFileElementType.kt
package io.stouder.adonis.edge

import com.intellij.psi.impl.source.html.HtmlFileImpl
import com.intellij.psi.stubs.PsiFileStub
import com.intellij.psi.tree.IStubFileElementType

class EdgeFileElementType private constructor() : IStubFileElementType<PsiFileStub<HtmlFileImpl>>("edge", EdgeLanguage.INSTANCE) {
    companion object {
        val INSTANCE: IStubFileElementType<PsiFileStub<HtmlFileImpl>> = EdgeFileElementType()
    }
}